lib/logger/lib/logger/watcher.ex

```diff
@@ -28,18 +26,24 @@ defmodule Logger.Watcher do
   Removes the given handler.
   """
   def unwatch(mod, handler) do
-    GenEvent.remove_handler(mod, handler, :ok)
+    result = :gen_event.delete_handler(mod, handler, :ok)
+    _ = Supervisor.terminate_child(@name, {__MODULE__, {mod, handler}})
+    result
```

Comment:
I am not sure if this could return strange results. For example `:gen_event.delete_handler/3` can return any result but `Supervisor.terminate_child/2` will still ensure the handler is stopped and not restarted.

Why not just use `terminate_child/2`? Our state is in the Supervisor and not the `:gen_event`.